modules/swift/examples/code_snippets/SampleCodeTest.swift

```diff
@@ -1209,2824 +1208,1603 @@ class SampleCodeTest {
+    
 }
 
-// end::sgw-repl-pull[]
+/* ----------------------------------------------------------- */
+/* ---------------------  PASSIVE SIDE  ---------------------- */
+/* ---------------  stubs for documentation  ----------------- */
+/* ----------------------------------------------------------- */
+class PassivePeerConnection: NSObject, MessageEndpointConnection {
+    
+    var messageEndpointListener: MessageEndpointListener?
+    var replicatorConnection: ReplicatorConnection?
+    
+    override init() {
+        super.init()
+    }
+    
+    func startListener() {
+        // tag::listener[]
+        let database = try! Database(name: "mydb")
+        let config = MessageEndpointListenerConfiguration(database: database, protocolType: .messageStream)
+        messageEndpointListener = MessageEndpointListener(config: config)
+        // end::listener[]
+    }
+    
+    func stopListener() {
+        // tag::passive-stop-listener[]
+        messageEndpointListener?.closeAll()
+        // end::passive-stop-listener[]
+    }
+    
+    func acceptConnection() {
+        // tag::advertizer-accept[]
+        let connection = PassivePeerConnection() /* implements MessageEndpointConnection */
+        messageEndpointListener?.accept(connection: connection)
+        // end::advertizer-accept[]
+    }
+    
+    func disconnect() {
+        // tag::passive-replicator-close[]
+        replicatorConnection?.close(error: nil)
+        // end::passive-replicator-close[]
+    }
+    
+    // tag::passive-peer-open[]
+    /* implementation of MessageEndpointConnection */
+    func open(connection: ReplicatorConnection, completion: @escaping (Bool, MessagingError?) -> Void) {
+        replicatorConnection = connection
+        completion(true, nil)
+    }
+    // end::passive-peer-open[]
+    
+    // tag::passive-peer-send[]
+    /* implementation of MessageEndpointConnection */
+    func send(message: Message, completion: @escaping (Bool, MessagingError?) -> Void) {
+        var data = Data()
+        data.append(message.toData())
+        /* send the data to the other peer */
+        /* ... */
+        /* call the completion handler once the message is sent */
+        completion(true, nil)
+    }
+    // end::passive-peer-send[]
+    
+    func receive(data: Data) {
+        // tag::passive-peer-receive[]
+        let message = Message.fromData(data)
+        replicatorConnection?.receive(message: message)
+        // end::passive-peer-receive[]
+    }
+    
+    // tag::passive-peer-close[]
+    /* implementation of MessageEndpointConnection */
+    func close(error: Error?, completion: @escaping () -> Void) {
+        /* disconnect with communications framework */
+        /* ... */
+        /* call completion handler */
+        completion()
+    }
+    // end::passive-peer-close[]
+}
+
+
+
+// MARK: -- QUESTIONS
 
-// tag::sgw-repl-pull-callouts[]
+// tag::listener[]
+// FIXME: Not sure why we need a tag like this???
```

Comment:
It is used here: https://docs.couchbase.com/couchbase-lite/current/swift/p2psync-custom.html#passive-peer

Reply:
added back!